packages/kbn-field-utils/src/utils/field_name_wildcard_matcher.ts

```diff
@@ -64,13 +115,12 @@ export function getFieldSearchMatchingHighlight(
   displayName: string,
   fieldSearchHighlight?: string
 ): string {
+  if (!fieldSearchHighlight) {
+    return '';
+  }
   const searchHighlight = (fieldSearchHighlight || '').trim();
-  if (
-    (searchHighlight.includes('*') || searchHighlight.includes(' ')) &&
-    fieldNameWildcardMatcher({ name: displayName }, searchHighlight)
-  ) {
-    return displayName;
+  if (displayName.toLowerCase().indexOf(searchHighlight.toLowerCase()) > -1) {
+    return searchHighlight;
   }
-
-  return searchHighlight;
+  return displayName;
```

Comment:
I think your choices for min length and distance make sense and work well overall, although I sometimes still saw results that felt unexpected, especially with deletions:
<img width="291" alt="fuzzy" src="https://github.com/elastic/kibana/assets/25592674/e167c807-42a6-4909-a9c8-478defb4c92f">

I think this tradeoff is ok since the fuzzy search is super useful, but it might help in these situations to highlight the substring that fuzzy matched instead of the whole display name. I tested this by returning the matching substring from `testFuzzySearchForString` and returning it in `getFieldSearchMatchingHighlight`:
```ts
return (
  testFuzzySearchForString(displayName.toLowerCase(), searchHighlight.toLowerCase()) ||
  displayName
);
```

Seems to work well and didn't feel like it had a performance impact with a 6000 field list. WDYT?
<img width="292" alt="fuzzy_highlight" src="https://github.com/elastic/kibana/assets/25592674/e8fdc7d5-6855-497b-86ae-765361e2a5f1">

Reply:
Fantastic idea Davis! I'll add it, thank you for spending time on it 🥇